domain/consensus/model/externalapi/block.go

```diff
@@ -63,9 +63,9 @@ func (header *DomainBlockHeader) Clone() *DomainBlockHeader {
 	return &DomainBlockHeader{
 		Version:              header.Version,
 		ParentHashes:         CloneHashes(header.ParentHashes),
```

Comment:
Is the call to CloneHashes necessary here?

Reply:
Purely theoretically, but yes - if you don't CloneHashes here, someone might change the contents of header.ParentHashes, and this will change the contents of the new block as well.